static/js/protein/data_processing_utils.ts

```diff
@@ -0,0 +1,448 @@
+  "GeneticVariantFunctionalCategoryCDSReference",
+  "GeneticVariantFunctionalCategoryncRNA",
+];
+
+const VARIANT_AFFECTS = [
+  "ClinSigAffects",
+  "ClinSigAssociation",
+  "ClinSigAssociationNotFound",
+  "ClinSigBenign",
+  "ClinSigBenignLikelyBenign",
+  "ClinSigConflictingPathogenicity",
+  "ClinSigDrugResponse",
+  "ClinSigHistocompatability",
+  "ClinSigLikelyBenign",
+  "ClinSigLikelyPathogenic",
+  "ClinSigNotProvided",
+  "ClinSigOther",
+  "ClinSigPathogenic",
+  "ClinSigPathogenicLikelyPathogenic",
+  "ClinSigProtective",
+  "ClinSigRiskFactor",
+  "ClinSigUncertain",
+  "ClinSigUnknown",
+  "ClinSigUntested",
+];
+
+const CHEM_RELATIONS = [
+  "RelationshipAssociationTypeAssociated",
+  "RelationshipAssociationTypeNotAssociated",
+  "RelationshipAssociationTypeAmbiguous",
+];
+
+export function getTissueScore(data: GraphNodes): ProteinStrData[] {
+  // Tissue to score mapping.
+  if (!data) {
+    return [];
+  }
+  const returnData: ProteinStrData[] = [];
+  // check for null values
+  if (_.isEmpty(data.nodes) || _.isEmpty(data.nodes[0].neighbors)) {
+    return [];
+  }
+  for (const neighbour of data.nodes[0].neighbors) {
+    if (neighbour.property !== "detectedProtein") {
+      continue;
+    }
+    for (const node of neighbour.nodes) {
+      let tissue = null;
+      let score = null;
+      // check for null or non-existent property values
+      if (_.isEmpty(node.neighbors)) {
+        continue;
+      }
+      for (const n of node.neighbors) {
+        if (n.property === "humanTissue") {
+          // check for null tissue value and move to next iteration if true
+          if (_.isEmpty(n.nodes[0].value)) {
+            continue;
+          }
+          tissue = n.nodes[0].value;
+        } else if (n.property === "proteinExpressionScore") {
+          score = n.nodes[0].value;
+        }
+      }
+      returnData.push({ name: tissue, value: score });
+    }
+    return returnData;
+  }
+  return [];
+}
+export function getProteinInteraction(
+  data: GraphNodes,
+  nodeName: string
+): InteractingProteinType[] {
+  // Protein Interaction to confidence score mapping.
+  if (!data) {
+    return [];
+  }
+  const returnData = [] as InteractingProteinType[];
+  const seen = new Set();
+  // check for null values
+  if (_.isEmpty(data.nodes) || _.isEmpty(data.nodes[0].neighbors)) {
+    return [];
+  }
+
+  for (const neighbour of data.nodes[0].neighbors) {
+    if (neighbour.property !== "interactingProtein") {
+      continue;
+    }
+    for (const node of neighbour.nodes) {
+      let protein_name = null;
+      let confidence_score = null;
+      let parent_protein = null;
+      // check for null or non-existent property values
+      if (_.isEmpty(node.neighbors)) {
+        continue;
+      }
+      for (const n of node.neighbors) {
+        if (n.property === "name") {
+          // check for null name values and move to next iteration if true
+          if (_.isEmpty(n.nodes[0].value)) {
```

Comment:
same here, check that n.nodes is not empty first

Reply:
Done!